samples/graphene-redis/Dockerfile

```
@@ -12,15 +12,26 @@ RUN make premain-graphene
 # Use with fully built graphene as build context
 # Place redis-server.manifest.template from this repo inside $GRAPHENE_DIR/Examples/redis/
 FROM ghcr.io/edgelesssys/edgelessrt-deploy:latest AS release
-RUN apt-get update && apt-get install -y build-essential autoconf gawk bison wget python3 libcurl4-openssl-dev \
-    python3-protobuf libprotobuf-c-dev protobuf-c-compiler python3-pip software-properties-common
+RUN apt-get update && apt-get install -y git meson build-essential autoconf gawk bison wget python3 libcurl4-openssl-dev \
+    python3-protobuf libprotobuf-c-dev protobuf-c-compiler python3-pip software-properties-common python3-click python3-jinja2 \
+    linux-headers-$(uname -r)
 RUN wget -qO- https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | apt-key add
 RUN add-apt-repository 'deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu bionic main'
 RUN apt-get install -y libsgx-quote-ex-dev libsgx-aesm-launch-plugin
-RUN python3 -m pip install toml
-COPY . /graphene
-COPY --from=build-premain /premain/build/premain-graphene /graphene/Examples/redis
+RUN python3 -m pip install "toml>=0.10"
+
+RUN git clone https://github.com/oscarlab/graphene.git /graphene
+WORKDIR /graphene
+RUN git reset --hard b37ac75efec0c1183fd42340ce2d3e04dcfb3388
+RUN openssl genrsa -3 -out Pal/src/host/Linux-SGX/signer/enclave-key.pem 3072
+RUN make ISGX_DRIVER_PATH=/usr/src/linux-headers-`uname -r`/arch/x86/ SGX=1
+RUN meson build -Ddirect=disabled -Dsgx=enabled
+RUN ninja -C build
+RUN ninja -C build install
+
+COPY --from=build-premain /premain/build/premain-graphene /graphene/Examples/redis/
+COPY redis-server.manifest.template /graphene/Examples/redis/
 WORKDIR /graphene/Examples/redis
 ENV BUILD_TLS yes
 RUN make clean && make SGX=1
-ENTRYPOINT [ "bash", "-c", "SGX=1 /graphene/Runtime/pal_loader /graphene/Examples/redis/redis-server" ]
+ENTRYPOINT [ "bash", "-c", "graphene-sgx /graphene/Examples/redis/redis-server" ]
```

```suggestion
ENTRYPOINT [ "graphene-sgx" "/graphene/Examples/redis/redis-server" ]
```
I would assume this to work now, unless `graphene-sgx` is only added to PATH through `.bashrc` or something similar

---

```
$ whereis graphene-sgx
graphene-sgx: /usr/local/bin/graphene-sgx
```

Should be in the default PATH, but otherwise we can also expect it to be in /usr/local/bin